### PR TITLE
Allow plugins to apply yaml

### LINKED ISSUE #1136: Allow plugins to apply yaml
Now that #1094 is merged.

Extend this functionality to plugins.

### PR DESCRIPTION
**Pull request by xtreme-vikram-yadav**
**What this PR does / why we need it**: 
- Allows GO plugins to apply yaml string to create/update resources.
- Improves consistency with JS plugins interface.

**Which issue(s) this PR fixes**
- Fixes #1136 

**Special notes for your reviewer**: 
Please merge after https://github.com/vmware-tanzu/octant/pull/2774 is merged.

### CODE REVIEW

**Comment by xtreme-vikram-yadav** [comment]
Using Update in api service interface to create and update seems a bit odd. This PR uses update to make the api consistent with JS plugins. It can be argued to have create and update but have similar signatures and underlying implementation which can help with clarity around developers intent.

**Comment by GuessWhoSamFoo** [comment]
The breaking changes are a bit too aggressive imo.

Create and Update has error handling for specific cases: e.g. create when the object already exists, update on an object which does not exist

Moreover, the preferred method of fetching objects is through `store.Get` which returns unstructured objects. The proposed changes here mean a conversion to YAML is required for each update operation.

In theory, one can already do the conversion from YAML via something like `k8s.io/client-go/kubernetes/scheme`. We could provide abstractions to help with this

**Comment by GuessWhoSamFoo** [comment]
I can see why typescript plugins have to do it this way because it is difficult to represent `unstructure.Unstructured` in https://github.com/vmware-tanzu/plugin-library-for-octant/blob/main/plugin/octant.d.ts#L143 , although in theory possible but hard to validate.

**Comment by xtreme-vikram-yadav** [comment]
Based on Sam's comments, a new method (`ApplyYaml`) is introduced in the dashboard client api leaving the `Create` and `Update` methods intact avoiding breaking api changes and providing better control for specific actions. Since a string yaml can also be applied using an internal action (https://github.com/vmware-tanzu/octant/blob/71e75e12b05764588d3f0aede455d143e35a3355/internal/octant/actions.go#L25), this will be documented for users.

File: web/src/stories/docs/plugins/GoPluginsIntro.story.mdx [hunk 1]
```diff
@@ -100,6 +100,19 @@ Here is an example of setting up your plugin to know when the current namespace
 	}
 ```
 
+Besides having custom actions, there are pre-exsiting octant actions which are registered by octant internal modules and they can be leveraged to perform an action.
+For example: `action.octant.dev/apply` (https://github.com/vmware-tanzu/octant/blob/71e75e12b05764588d3f0aede455d143e35a3355/internal/octant/actions.go#L25) can be used to create/update resources
```
[the remark this answers is not in the document]
Comment (xtreme-vikram-yadav) [reply web/src/stories/docs/plugins/GoPluginsIntro.story.mdx:104]: I apologize I forgot to comment here. I am a bit confused about this. There are multiple actions which allow users to interact with the resources in `octant/actions`. Should we also move other actions to `pkg/actions` as well as part of this documentation? Or may be the path forward is to have make needed functionality available through plugin API with some duplication?

[the remark this answers is not in the document]
Comment (wwitzel3) [reply web/src/stories/docs/plugins/GoPluginsIntro.story.mdx:104]: Yeah, I think there is some transition cruft here and that we want to expose actions for users to consume via the `pkg` vs. `internal`

**Comment by GuessWhoSamFoo** [comment]
Some notes:

```
res, err := request.DashboardClient.ApplyYAML(request.Context(), "default", "apiVersion: v1\nkind: Pod\nmetadata:\n  name: static-web\n  labels:\n    role: myrole\nspec:\n  containers:\n    - name: web\n      image: nginx\n      ports:\n        - name: web\n          containerPort: 80\n          protocol: TCP\n")
if err != nil {
	return err
}
```

This method of triggering an action requires binding to a component and an understanding of what payload metadata is needed since there is no `RunAction` rpc call
```
button := component.NewButton("Create pod", action.Payload{"action": action.ActionApplyYaml, "namespace": "default", "update": "apiVersion: v1\nkind: Pod\nmetadata:\n  name: static-web\n  labels:\n    role: myrole\nspec:\n  containers:\n    - name: web\n      image: nginx\n      ports:\n        - name: web\n          containerPort: 80\n          protocol: TCP\n"})
```

`ApplyYAML` returns a slice of strings that seems to be a tokenized message of the server response (e.g. `[Updated Pod (v1) static-web in default]`) although not documented anywhere

**Review by GuessWhoSamFoo — APPROVED** [review]

**Pull request merged**